proposals/NNNN-service-context.md

```diff
@@ -0,0 +1,177 @@
+# Swift Service Context
+
+* Proposal: [SSWG-NNNN](NNNN-service-context.md)
+* Authors: [Konrad 'ktoso' Malawski](https://github.com/ktoso) / [Moritz Lang](https://github.com/slashmo)
+* Review Manager: TBD
+* Status: **Implemented**
+* Implementation: 
+  * [swift-service-context](https://github.com/apple/swift-service-context)
+* Forum Threads: TBD
+
+## Package Description
+
+ServiceContext is a minimal (zero-dependency) context propagation container, intended to "carry" items for purposes of cross-cutting tools to be built on top of it.
+
+It is modeled after the concepts explained in W3C Baggage and the in the spirit of Tracing Plane's "Baggage Context" type, although by itself it does not define a specific serialization format.
+
+See https://github.com/apple/swift-distributed-tracing for actual instrument types and implementations which can be used to deploy various cross-cutting instruments all reusing the same baggage type. More information can be found in the SSWG meeting notes.
+
+| | |
+|--|--|
+| **Package Name** | `swift-service-context` |
+| **Module Name** | `ServiceContextModule` |
+| **Proposed Maturity Level** | [Incubating](https://github.com/swift-server/sswg/blob/main/process/incubation.md#process-diagram) |
+| **License** | [Apache v2](https://github.com/apple/swift-distributed-tracing/blob/main/LICENSE.txt) |
+| **Dependencies** | [swift-service-context](https://github.com/apple/swift-service-context) |
+
+## Introduction
+
+Service context is a type-safe dictionary that is keyed using types and intended to be used for contextual information propagation across asynchronous code.
+
+It declares a well-known Task Local key for the service context, and patterns how to add keys. 
+
+It is a core building block of [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) which uses it to propagate trace information across swift concurrency tasks, though can be used independently as well.
```

Comment:
```suggestion
It is a core building block of [swift-distributed-tracing](https://github.com/apple/swift-distributed-tracing) which uses it to propagate trace information across swift concurrency tasks, though it can be used independently as well.
```